### DISCUSSION TITLE
How to headless install ? (Ubuntu Server)

### DISCUSSION THREAD

**Discussion by Adanorm** [FoldingAtHome/fah-client-bastet#214] [General]
Hello,

I'm looking to migrate a first "real" machine from v7 to v8.3.4.
I think v8.3.4 is mature for this first try outside of a small VM. It will allow me to test the GPU part of the v8.

This PC is running headless with Ubuntu Server 22.04. But this new client requiers a graphical session+browser to connect to the account.
Someone said me before it was possible, I lost the link. Can you help me ?
Thanks in advance,

**Comment by jcoffland** [comment]
Here's the procedure:

* Register a F@H account if you don't already have one.  https://beta.foldingathome.org/
* Login in to your F@H account.
* Go to your account settings and copy the current account token.
* Install fah-client on the target machine as normal.
* Edit ``/etc/fah-client/config.xml`` and add the following:
```xml
<config>
  <account-token v="<your account token>"/>
  <machine-name v="<a good display name for this machine>"/>
</config>
```
* Then restart your client with ``sudo systemctl restart fah-client``
* Wait a few seconds then reload Web Control where you logged in.  The new machine should appear there.

Note, you can leave off ``machine-name`` and it will default to the machine's hostname.  You also can leave the ``account-token`` in the config or take it out.  Once the client is connected it is no longer needed.  You can also generate a new token in your account if you want to make sure no one else can use it to join clients to your account.

If it doesn't work check the client's logs in ``/var/log/fah-client/`` for errors.

**Reply by Adanorm** [reply]
I can't find the account settings. Nothing on the "head" of the page, nothing on the 4 tabs.

**Reply by Adanorm** [reply]
I wrote this : https://beta.foldingathome.org/account, it did the trick !

**Reply by Kalusd** [reply]
Hi, I first tried to install Folding@Home in a Docker container but it did not work. So I tried to install it directly on my machine, which is on Ubuntu Server 22.04. Since I use SSH to manage this computer, I followed all your instructions to install it without a graphical interface. But I keep running into the same problem as in the docker container :

```
02:04:37:I1:OUT10:> PUT https://api.foldingathome.org/machine/jK8j1A... HTTP/1.1
02:04:37:E :OUT10:Failed response: EOF
02:05:41:I1:OUT11:> GET https://api.foldingathome.org/gpus HTTP/1.1
02:05:41:W :Failed to update GPUs
02:05:52:I1:OUT12:> PUT https://api.foldingathome.org/machine/jK8j1A... HTTP/1.1
02:05:52:E :OUT12:Failed response: EOF
02:07:45:I1:OUT13:> PUT https://api.foldingathome.org/machine/jK8j1A... HTTP/1.1
02:07:45:E :OUT13:Failed response: EOF
02:08:53:I1:OUT14:> GET https://api.foldingathome.org/gpus HTTP/1.1
02:08:54:W :Failed to update GPUs
02:10:36:I1:OUT15:> PUT https://api.foldingathome.org/machine/jK8j1A... HTTP/1.1
02:10:36:E :OUT15:Failed response: EOF
```

I also can't ping api.foldingathome.org, I get Hostname: Name or service not known

Thanks for your answer.

**Reply by kbernhagen** [reply]
You're having general networking problems, for which I can't help.

For a v8 docker container, you may be interested in
https://github.com/linuxserver/docker-foldingathome

**Reply by Kalusd** [reply]
I think that's the one I already tried, but I'll try again and ask there if someone had the same problem.
Thanks for your answer

**Reply by Kalusd** [reply]
I did the same thing as before but this time it seems to be working, maybe api.foldingathome.org was overloaded.

Edit : I now have a 503 on the various assignment server but I can access them on another machine

```
14:14:20:I1:OUT9:> POST https://assign5.foldingathome.org/api/assign HTTP/1.1
14:14:20:I1:OUT9:< HTTP/1.1 503 HTTP_SERVICE_UNAVAILABLE
14:14:20:E :OUT9:HTTP_SERVICE_UNAVAILABLE: {"error":"No appropriate assignment"}
14:14:20:I1:WU2:Retry #5 in 32 secs
14:14:52:I1:WU2:Requesting WU assignment for user ... team ...
14:14:52:I1:OUT10:> POST https://assign6.foldingathome.org/api/assign HTTP/1.1
14:14:53:I1:OUT10:< HTTP/1.1 503 HTTP_SERVICE_UNAVAILABLE
14:14:53:E :OUT10:HTTP_SERVICE_UNAVAILABLE: {"error":"No appropriate assignment"}
14:14:53:I1:WU2:Retry #6 in 1 min 4 secs
14:15:57:I1:WU2:Requesting WU assignment for user ... team ...
14:15:57:I1:OUT11:> POST https://assign1.foldingathome.org/api/assign HTTP/1.1
14:15:58:I1:OUT11:< HTTP/1.1 503 HTTP_SERVICE_UNAVAILABLE
14:15:58:E :OUT11:HTTP_SERVICE_UNAVAILABLE: {"error":"No appropriate assignment"}
14:15:58:I1:WU2:Retry #7 in 2 mins 8 secs
```
It seems that the work server with projects that will assign to CPU thread counts less than 4 is down.

Thanks for your help, have a great day

**Reply by tabasavr** [reply]
I'm also getting EOF error on 8.3.18
```
07:59:12:I1:OUT7:> PUT https://api.foldingathome.org/machine/REDACTED HTTP/1.1
07:59:12:E :OUT7:Failed response: EOF
```
When I try to run the same PUT request via Firefox devtools I get status code 500 HTTP_INTERNAL_SERVER_ERROR

**Reply by tabasavr** [reply]
Got same error with 8.4.4 built from source. `ping  api.foldingathome.org` works fine, so it's not a network issue. Hopefully #273 will make this error clearer

OS: Ubuntu 24.04.1 LTS x86_64
Kernel: 6.8.0-41-generic

**Comment by jcoffland** [comment]
Note, passing the account token is unnecessary if you login to F@H with a browser on the same machine where the fah-client is running.  This is because Web Control tries to connect to any locally running fah-client.  If it finds one and that client has not already joined another F@H account, then Web Control will transparently pass the current account token and the local client will automatically join the F@H account.

But of course, you asked about a headless install.

**Comment by jcoffland** [comment]
You can get to the account settings by clicking the user icon in the top right corner of Web Control.

**Comment by CHDAFNI-MSFT** [comment]
@jcoffland  - as someone who is also trying to consider/setup a large number of headless machines - the https://beta.foldingathome.org/ site doesn't appear to visually show a way to locate or otherwise navigate to the account settings page.

I ended up following @Adanorm link to obtain it.
![Screenshot 2024-06-03 131824](https://github.com/FoldingAtHome/fah-client-bastet/assets/28310425/11d56f97-a111-49be-979e-7630b83e9c17)

**Comment by jcoffland** [comment]
I'm pretty sure the problem is that your browser has cached the old Web Control.  The newer versions of Web Control detect this situation automatically but there no solution but to do a hard page reload when coming from versions prior to v8.3.x.

**Reply by CHDAFNI-MSFT** [reply]
@jcoffland  - just know that even from an incognito window or mobile browser - I get the same experience to the screenshot I posted earlier. 

I'm curious what you see or what I am supposed to see should I be hitting the new Web Control.

**Comment by kbernhagen** [comment]
When logged in:
![image](https://github.com/FoldingAtHome/fah-client-bastet/assets/6762784/55023a61-563f-41c5-80d5-86373560efaf)

When not logged in:
![image](https://github.com/FoldingAtHome/fah-client-bastet/assets/6762784/86311899-1016-4716-8a6f-a4eb4c0aa6fd)

**Reply by CHDAFNI-MSFT** [reply]
Works for Alpha. Good enough for me. Thanks for the pointer

**Comment by kbernhagen** [comment]
Note, when using an alpha release of the client, you should use
https://alpha.foldingathome.org/

**Reply by kbernhagen** [reply]
Alpha v8.4 Web Control is
https://v8-4.foldingathome.org/
